packages/prepo-utils/__tests__/utils.test.ts

```diff
@@ -37,3 +45,72 @@ describe('prepo-utils', () => {
     expect(formattedNumber3).toEqual('$123M')
   })
 })
+
+describe('# parseUnits', () => {
+  // test decimals
+  it('returns undefined if decimals undefined', () => {
+    const output = parseUnits('100', undefined)
+    expect(output).toBe(undefined)
+  })
+
+  it('returns correct BigNumber if valid decimals', () => {
+    validDecimals.forEach(({ input, decimals, output }) => {
+      const result = parseUnits(input, decimals)
+      expect(result.eq(output)).toBe(true)
+    })
+  })
+
+  // test integer only value
+  it('returns 0 BigNumber if empty string', () => {
+    const output = parseUnits('', 6)
+    expect(output.eq(BigNumber.from(0))).toBe(true)
+  })
+
+  it('returns correct BigNumber if valid integer only input', () => {
+    validIntegerInput.forEach(({ input, decimals, output }) => {
+      const result = parseUnits(input, decimals)
+      expect(result.eq(output)).toBe(true)
+    })
+  })
+
+  it('returns undefined if invalid integer number', () => {
+    const output = parseUnits('abcd1234', 6)
+    expect(output).toBe(undefined)
+  })
+
+  // test fraction only value
+  it('returns correct BigNumber if valid fraction only input', () => {
+    validFractionInput.forEach(({ input, decimals, output }) => {
+      const result = parseUnits(input, decimals)
+      expect(result.eq(output)).toBe(true)
+    })
+  })
+
+  it('returns BigNumber capped at specified decimals if more fraction digits than decimals', () => {
+    const output = parseUnits('0.1234567890', 6)
+    expect(output.eq(BigNumber.from(123456))).toBe(true)
+  })
+
+  it('returns undefined if fraction contains non-number character', () => {
+    const output = parseUnits('0.1234567890a', 6)
+    expect(output).toBe(undefined)
+  })
+
+  // test integer + fraction part together
+  it('returns correct BigNumber if valid full input', () => {
+    validFullInput.forEach(({ input, decimals, output }) => {
+      const result = parseUnits(input, decimals)
+      expect(result.eq(output)).toBe(true)
+    })
+  })
+
+  it('returns undefined if value 12,34', () => {
```

Comment:
```suggestion
  it('returns undefined if comma in integer part', () => {
```
and group with integer part tests